### ISSUE TITLE
No module jwt

### ISSUE THREAD

**Issue by pdelboca** [datopian/ckanext-authz-service#3]
``` 
 File "/srv/app/src_extensions/ckanext-authz-service/ckanext/authz_service/actions.py", line 8, in <module>
    import jwt
ImportError: No module named jwt
```
When installing the extension, this module is not found. It is missing in the requirements files.

**Comment by shevron** [comment]
Make sure you installed the rights `requirements.*.txt` file for your environment. You can easily do this by going into your virtual environment, then running:

    cd path/to/ckanext-authz-service
    make install

Or you can simply run:

    pip install -r path/to/ckan-authz-service/requirements.py2.txt

If you are on Python 2, or `requirements.py3.txt` if on Python 3.

**Comment by shevron** [comment]
@pdelboca I'm closing this, please reopen if still relevant

**Comment by mbeilin** [comment]
@shevron the same issue for me here (developer installation)
```
ckan_1   | Traceback (most recent call last):
ckan_1   |   File "/usr/bin/paster", line 11, in <module>
ckan_1   |     load_entry_point('PasteScript', 'console_scripts', 'paster')()
ckan_1   |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 102, in run
ckan_1   |     invoke(command, command_name, options, args[1:])
ckan_1   |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 141, in invoke
ckan_1   |     exit_code = runner.run(args)
ckan_1   |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 236, in run
ckan_1   |     result = self.command()
ckan_1   |   File "/usr/lib/python2.7/site-packages/paste/script/serve.py", line 284, in command
ckan_1   |     relative_to=base, global_conf=vars)
ckan_1   |   File "/usr/lib/python2.7/site-packages/paste/script/serve.py", line 329, in loadapp
ckan_1   |     **kw)
ckan_1   |   File "/usr/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 247, in loadapp
ckan_1   |     return loadobj(APP, uri, name=name, **kw)
ckan_1   |   File "/usr/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 272, in loadobj
ckan_1   |     return context.create()
ckan_1   |   File "/usr/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 710, in create
ckan_1   |     return self.object_type.invoke(self)
ckan_1   |   File "/usr/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 146, in invoke
ckan_1   |     return fix_call(context.object, context.global_conf, **context.local_conf)
ckan_1   |   File "/usr/lib/python2.7/site-packages/paste/deploy/util.py", line 55, in fix_call
ckan_1   |     val = callable(*args, **kw)
ckan_1   |   File "/srv/app/src/ckan/ckan/config/middleware/__init__.py", line 55, in make_app
ckan_1   |     load_environment(conf, app_conf)
ckan_1   |   File "/srv/app/src/ckan/ckan/config/environment.py", line 116, in load_environment
ckan_1   |     p.load_all()
ckan_1   |   File "/srv/app/src/ckan/ckan/plugins/core.py", line 140, in load_all
ckan_1   |     load(*plugins)
ckan_1   |   File "/srv/app/src/ckan/ckan/plugins/core.py", line 154, in load
ckan_1   |     service = _get_service(plugin)
ckan_1   |   File "/srv/app/src/ckan/ckan/plugins/core.py", line 256, in _get_service
ckan_1   |     return plugin.load()(name=plugin_name)
ckan_1   |   File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2443, in load
ckan_1   |     return self.resolve()
ckan_1   |   File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2449, in resolve
ckan_1   |     module = __import__(self.module_name, fromlist=['__name__'], level=0)
ckan_1   |   File "/srv/app/src_extensions/ckanext-authz-service/ckanext/authz_service/plugin.py", line 5, in <module>
ckan_1   |     from ckanext.authz_service import actions
ckan_1   |   File "/srv/app/src_extensions/ckanext-authz-service/ckanext/authz_service/actions.py", line 8, in <module>
ckan_1   |     import jwt
ckan_1   | ImportError: No module named jwt

```

**Comment by shevron** [comment]
@mbeilin please provide the output of `pip freeze`, `python --version` and `echo $VIRTUAL_ENV` from within your virtual environment.

**Comment by mbeilin** [comment]
@shevron please see below the required info:
1. pip freeze
```
alabaster==0.7.12
atomicwrites==1.3.0
attrs==19.3.0
Babel==2.3.4
backports.functools-lru-cache==1.6.1
Beaker==1.9.0
beautifulsoup4==4.5.1
bleach==2.1.3
blinker==1.4
certifi==2020.4.5.1
cffi==1.14.0
chardet==3.0.4
-e git+https://github.com/ckan/ckan.git@8e1cc60b2fa11da6843051678b7ee2cc08c2a7a9#egg=ckan
-e git+git@github.com:datopian/ckanext-authz-service.git@ed63cd303436df999bf4f124c051261936d68871#egg=ckanext_authz_service
-e git+https://github.com/datopian/ckanext-external-storage.git@918524d2cf35e8f6e19836e14064a76d1bad1f1b#egg=ckanext_external_storage
click==7.1.1
configparser==4.0.2
contextlib2==0.6.0.post1
coverage==4.5.4
coveralls==1.8.2
cryptography==2.9.2
decorator==4.2.1
docopt==0.6.2
docutils==0.12
entrypoints==0.3
enum34==1.1.10
extras==1.0.0
factory-boy==2.1.1
fanstatic==0.12
first==2.0.2
fixtures==3.0.0
flake8==3.7.9
Flask==0.12.4
Flask-Babel==0.11.2
Flask-DebugToolbar==0.10.1
FormEncode==1.3.1
funcsigs==1.0.2
functools32==3.2.3.post2
futures==3.3.0
html5lib==1.0.1
httpretty==0.8.3
idna==2.9
imagesize==1.2.0
importlib-metadata==1.6.0
ipaddress==1.0.23
isort==4.3.21
itsdangerous==0.24
Jinja2==2.8
linecache2==1.0.0
Mako==1.0.7
Markdown==2.6.7
MarkupSafe==1.0
mccabe==0.6.1
mock==2.0.0
more-itertools==5.0.0
mox3==0.28.0
nose==1.3.7
ofs==0.4.2
packaging==20.3
Pairtree===0.7.1-T
passlib==1.6.5
Paste==1.7.5.1
PasteDeploy==1.5.2
PasteScript==2.0.2
pathlib2==2.3.5
pbr==1.10.0
pip-tools==4.5.1
pluggy==0.13.1
polib==1.0.7
psycopg2==2.7.3.2
py==1.8.1
pycodestyle==2.5.0
pycparser==2.20
pyfakefs==2.9
pyflakes==2.1.1
Pygments==2.2.0
PyJWT==1.7.1
Pylons==0.9.7
pyOpenSSL==19.1.0
pyparsing==2.4.7
pysolr==3.6.0
pytest==4.6.9
pytest-ckan==0.0.12
pytest-cov==2.8.1
pytest-flake8==1.0.5
pytest-isort==0.3.1
python-dateutil==1.5
python-magic==0.4.15
python-mimeparse==1.6.0
pytz==2020.1
pyutilib.component.core==4.6.4
PyYAML==5.3.1
redis==2.10.6
repoze.lru==0.7
repoze.who==2.3
repoze.who-friendlyform==1.0.8
requests==2.23.0
Routes==1.13
rq==0.6.0
scandir==1.10.0
simplejson==3.10.0
six==1.14.0
snowballstemmer==2.0.0
Sphinx==1.7.1
sphinx-rtd-theme==0.3.1
sphinxcontrib-websupport==1.1.2
SQLAlchemy==1.1.11
sqlalchemy-migrate==0.10.0
sqlparse==0.2.2
Tempita==0.5.2
testtools==2.4.0
traceback2==1.4.0
typing==3.7.4.1
typing-extensions==3.7.4.2
tzlocal==1.3
unicodecsv==0.14.1
unittest2==1.1.0
urllib3==1.25.9
vdm==0.14
wcwidth==0.1.9
webencodings==0.5.1
WebError==0.13.1
WebHelpers==1.3
WebOb==1.0.8
WebTest==1.4.3
Werkzeug==0.14.1
zipp==1.2.0
zope.interface==4.3.2
```

2. python --version
`Python 2.7.17`

3. echo $VIRTUAL_ENV
`/home/michael/PycharmProjects/nhs-ckan-docker/venv`

**Comment by shevron** [comment]
@mbeilin your virtual environment that you are in (`/home/michael/PycharmProjects/nhs-ckan-docker/venv`) has PyJWT installed, but you are probably not using this virtual environment when running CKAN or whatever command you ran that generated the backtrace above. Note that, for example, the path to `paster` in the backtrace is `/usr/bin/paster` i.e. your global Python installation. 

I don't know how your local environment is set up, but the Python environment that runs CKAN should be the environment into which you install ckanext-authz-service and all it's dependencies. It can either be a virtual environment (recommended) or a global environment, but it has to be the same one. 

Of course, if CKAN is in some kind of Docker container, the dependencies need to be installed there. 

In short, this is most likely not a bug in ckanext-authz-service and just an environment inconsistency issue. Please check how your environment is set up, and ping me in a DM if more assistance is needed.
